x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/execution/assembler/CircuitBreakerTests.java

```diff
@@ -0,0 +1,111 @@
+
+    private final List<HitExtractor> keyExtractors = emptyList();
+    private final HitExtractor tsExtractor = TimestampExtractor.INSTANCE;
+    private final HitExtractor implicitTbExtractor = ImplicitTiebreakerHitExtractor.INSTANCE;
+    private final int stages = randomIntBetween(3, 10);
+
+    static class TestQueryClient implements QueryClient {
+
+        @Override
+        public void query(QueryRequest r, ActionListener<SearchResponse> l) {
+            int ordinal = r.searchSource().terminateAfter();
+            SearchHit searchHit = new SearchHit(ordinal, String.valueOf(ordinal), null, null);
+            searchHit.sortValues(new SearchSortValues(
+                    new Long[] { (long) ordinal, 1L },
+                    new DocValueFormat[] { DocValueFormat.RAW, DocValueFormat.RAW }));
+            SearchHits searchHits = new SearchHits(new SearchHit[] { searchHit }, new TotalHits(1, Relation.EQUAL_TO), 0.0f);
+            SearchResponseSections internal = new SearchResponseSections(searchHits, null, null, false, false, null, 0);
+            SearchResponse s = new SearchResponse(internal, null, 0, 1, 0, 0, null, Clusters.EMPTY);
+            l.onResponse(s);
+        }
+
+        @Override
+        public void fetchHits(Iterable<List<HitReference>> refs, ActionListener<List<List<SearchHit>>> listener) {
+            List<List<SearchHit>> searchHits = new ArrayList<>();
+            for (List<HitReference> ref : refs) {
+                List<SearchHit> hits = new ArrayList<>(ref.size());
+                for (HitReference hitRef : ref) {
+                    hits.add(new SearchHit(-1, hitRef.id(), null, null));
+                }
+                searchHits.add(hits);
+            }
+            listener.onResponse(searchHits);
+        }
+    }
+
+    public void testCircuitBreaker() {
+        QueryClient client = new TestQueryClient();
+        List<Criterion<BoxedQueryRequest>> criteria = new ArrayList<>(stages);
+
+        for (int i = 0; i < stages; i++) {
+            final int j = i;
+            criteria.add(new Criterion<>(i,
+                    new BoxedQueryRequest(() -> SearchSourceBuilder.searchSource()
+                            .size(10)
+                            .query(matchAllQuery())
+                            .terminateAfter(j), "@timestamp", emptyList()),
+                    keyExtractors,
+                    tsExtractor,
+                    null,
+                    implicitTbExtractor,
+                    false));
+        }
+
+        SequenceMatcher matcher = new SequenceMatcher(stages, false, TimeValue.MINUS_ONE, null, CIRCUIT_BREAKER);
+        TumblingWindow window = new TumblingWindow(client, criteria, null, matcher);
+        window.execute(wrap(p -> {}, ex -> {
+            throw ExceptionsHelper.convertToRuntime(ex);
+        }));
+
+        CIRCUIT_BREAKER.startBreaking();
+        window.execute(wrap(p -> {}, ex -> assertEquals(CircuitBreakingException.class, ex.getClass())));
```

Comment:
I think this test would also pass if the circuit breaker never throws / is never called.

Reply:
Thx! Fixed with `expectThrows()`